torchci/lib/bot/pytorchBotHandler.ts

```diff
@@ -141,20 +146,34 @@ The explanation needs to be clear on why this is needed. Here are some good exam
     return null;
   }
 
-  async doesHaveApprovedStatus() {
-    var res = await this.ctx.octokit.pulls.listReviews({
-      owner: this.owner,
-      repo: this.repo,
-      pull_number: this.prNum,
-    })
+  async getApprovalStatus(): Promise<string> {
+    var reviews:PullRequestReview[] = []
+    const REVIEWS_PER_PAGE = 100
+
+    var page = 1;
+    var gotAllPages = false
+    while (!gotAllPages) {
+      var res = await this.ctx.octokit.pulls.listReviews({
+        owner: this.owner,
+        repo: this.repo,
+        pull_number: this.prNum,
+        per_page: REVIEWS_PER_PAGE,
+        page: page
+      })
+      
+      if ((res?.data?.length ?? 0) < REVIEWS_PER_PAGE) {
```

Comment:
I thought I commented on this already, but apparently it didn't save...
There should be a way to get if it's the last page explicitly, no?
The current implementation of `gotAllPages` will not be correct if there are exactly 100 (or 200, etc.) reviews.

Reply:
It would have resulted in an extra request which returned zero items. But you're right, there's a better API available. Switching over to that...